### PR TITLE
Fixing ESLint in docs. (5/n, 100% done)

### PR DESCRIPTION
**Pull request by anthonyshew**
## Description
4/n of my stacked diffs to address ESLint errors in docs. PRs to ship before this one:
- https://github.com/vercel/turbo/pull/6812
- #6813
- #6818 
- #6819 

There are a lot of file changes on this one just because there are component export renames.

### CODE REVIEW

File: docs/.eslintrc.js [hunk 1]
```diff
@@ -18,36 +18,6 @@ module.exports = {
     React: true,
     NodeJS: true,
   },
-  rules: {
```
Comment (anthonyshew) [inline docs/.eslintrc.js:21]: LFG

File: docs/components/Authors.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import type { Author } from "../content/team";
 import { Avatar } from "./Avatar";
 
 export function Authors({ authors }: { authors: Author[] }) {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- This does what is is meant to?
```
Comment (anthonyshew) [inline docs/components/Authors.tsx:7]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- This does what is is meant to.
```

File: docs/package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "dev": "next",
     "start": "next start",
     "build": "next build ",
-    "lint": "eslint . --max-warnings 40",
+    "lint": "eslint . --max-warnings 0",
```
Comment (anthonyshew) [inline docs/package.json:10]: LFG

Comment (tknickman) [reply docs/package.json:10]: Let's re-enable lint check in the next.config.json as well!

**Review by tknickman — APPROVED** [review]

File: docs/package.json (hunk 3 above)
Comment (anthonyshew) [reply docs/package.json:10]: After chatting, we realized we don't need it. A GitHub Action does the check already.

**Pull request merged**